src/components/ComputeModal.js

```diff
@@ -0,0 +1,1335 @@
+    return desiredRuntime.cloudService === cloudServices.DATAPROC && hasAttachedDisk()
+  }
+
+  const willRequireDowntime = () => {
+    const { runtime: existingRuntime } = getExistingEnvironmentConfig()
+    return existingRuntime && (!canUpdateRuntime() || isStopRequired())
+  }
+  // Helper functions -- end
+
+  // Lifecycle
+  Utils.useOnMount(() => {
+    // Can't pass an async function into useEffect so we define the function in the body and then call it
+    const doUseOnMount = _.flow(
+      withErrorReporting('Error loading cloud environment'),
+      Utils.withBusyState(setLoading)
+    )(async () => {
+      const { googleProject } = getWorkspaceObject()
+      const currentRuntime = getCurrentRuntime(runtimes)
+      const currentPersistentDisk = getCurrentPersistentDisk(runtimes, persistentDisks)
+
+      Ajax().Metrics.captureEvent(Events.cloudEnvironmentConfigOpen, {
+        existingConfig: !!currentRuntime, ...extractWorkspaceDetails(getWorkspaceObject())
+      })
+      const [currentRuntimeDetails, newLeoImages, currentPersistentDiskDetails] = await Promise.all([
+        currentRuntime ? Ajax().Runtimes.runtime(currentRuntime.googleProject, currentRuntime.runtimeName).details() : null,
+        Ajax()
+          .Buckets
+          .getObjectPreview('terra-docker-image-documentation', 'terra-docker-versions.json', googleProject, true)
+          .then(res => res.json()),
+        currentPersistentDisk ? Ajax().Disks.disk(currentPersistentDisk.googleProject, currentPersistentDisk.name).details() : null
+      ])
+      const filteredNewLeoImages = !!tool ? _.filter(image => _.includes(image.id, tools[tool].imageIds), newLeoImages) : newLeoImages
+
+      const imageUrl = currentRuntimeDetails ? getImageUrl(currentRuntimeDetails) : _.find({ id: 'terra-jupyter-gatk' }, newLeoImages).image
+      const foundImage = _.find({ image: imageUrl }, newLeoImages)
+
+      /* eslint-disable indent */
+        // TODO: open to feedback and still thinking about this...
+        // Selected Leo image uses the following logic (psuedoCode not written in same way as code for clarity)
+        // if found image (aka image associated with user's runtime) NOT in newLeoImages (the image dropdown list from bucket)
+        //   user is using custom image
+        // else if found Image NOT in filteredNewLeoImages (filtered based on analysis tool selection) and isAnalysisMode
+        //   use default image for selected tool
+        // else
+        //   use imageUrl derived from users current runtime
+        /* eslint-disable indent */
+        const getSelectedImage = () => {
+          if (foundImage) {
+            if (!_.includes(foundImage, filteredNewLeoImages) && isAnalysisMode) {
+              return _.find({ id: tools[tool].defaultImageId }, newLeoImages).image
+            } else {
+              return imageUrl
+            }
+          } else {
+            return customMode
+          }
+        }
+
```

Comment:
The states updates below used to be also made in `NewRuntimeModal`'s `constructor` (in addition to `componentDidMount` which is the class component equivalent of `useEffect` that `useOnMount` uses under the hood). This seemed redundant as the modal is mounted quickly. Not sure if it may be undesirable in case of major network latencies.